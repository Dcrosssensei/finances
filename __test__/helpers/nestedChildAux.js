export const getNestedChild = (obj, depth) => {
    if (depth === 0 || !obj?.children) return obj;
    return getNestedChild(obj.children[0], depth - 1);
  }
  
  