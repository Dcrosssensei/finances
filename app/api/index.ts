export * from './helpers/transforClass';
export * from './operations';
export * from './validation';
