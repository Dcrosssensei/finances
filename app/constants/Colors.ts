
export const ColorsApp={
  base: '#fff',
  blue: '#41537d',
  gray:"#636262",
  lightgray: "#bfbfbf",
  softGray: '#edf0f2',
  red: '#e60023',
  yellow: '#ffd100',
  darkyellow: '#ffb900'
}