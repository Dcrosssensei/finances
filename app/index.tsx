import { createStackNavigator } from '@react-navigation/stack';
import Home from "./pages/home/Home";
import Details from './pages/details/Details';
import { RootStackParamList } from './routes/navigationTypes';
import Header from './components/header/Header';
import { RecordProvider } from './context/MainContext';

const Stack = createStackNavigator<RootStackParamList>();


export default function Index() {

  return (
    <RecordProvider>
      <Stack.Navigator>
        <Stack.Screen name="Home" component={Home} options={{ headerShown: false }} />
        <Stack.Screen name="Details" component={Details} 
        options={ {
          header: ()=> (<Header back />)
        }}  
        />
      </Stack.Navigator>
    </RecordProvider>
  );
}
