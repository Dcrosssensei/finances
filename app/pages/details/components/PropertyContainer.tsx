import { StyleSheet, Text, View } from 'react-native'
import React, { ReactNode } from 'react'


type PropertyContainerProps = {
    children: ReactNode;
  };
  
  const PropertyContainer = ({ children }: PropertyContainerProps) => {
    return (
      <View style={style.propertiesContainer}>
        {children}
      </View>
    )
  }

export default PropertyContainer

const style = StyleSheet.create({
    propertiesContainer: {
        flexDirection: 'row',
        justifyContent: 'space-between',
        paddingVertical: 10
      },
})