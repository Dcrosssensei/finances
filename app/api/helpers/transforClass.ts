import { ProductTypes, ProductTypesToSend } from '@/app/models';
import { format } from 'date-fns';

export function transformToSend(data: ProductTypes): ProductTypesToSend {
  return {
    ...data, 
    date_release: format(data.date_release, 'yyyy-MM-dd'), 
    date_revision: format(data.date_revision, 'yyyy-MM-dd') 
  };
}
