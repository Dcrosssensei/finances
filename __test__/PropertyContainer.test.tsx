import { ColorsApp } from "@/app/constants";
import PropertyContainer from "@/app/pages/details/components/PropertyContainer";
import React from "react";
import { Text, View } from "react-native";
import renderer from "react-test-renderer";
import { getNestedChild } from "./helpers/nestedChildAux";


describe('PropertyContainer', () => {
    let tree: any
    beforeAll(()=>{
        tree = renderer.create(
        <PropertyContainer >
              <View >
                <Text>Soy un texto</Text>
              </View>
        </PropertyContainer>
        ).toJSON();
    })
    it('renders correctly', () => {
        expect(tree).toMatchSnapshot();
      });

    it('PropertyContainer render text', () => {
        const textRendered = getNestedChild(tree,3);  
        expect(textRendered).toEqual("Soy un texto");
      });

  });

