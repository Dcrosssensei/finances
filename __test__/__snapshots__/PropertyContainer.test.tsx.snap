// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PropertyContainer renders correctly 1`] = `
<View
  style={
    {
      "flexDirection": "row",
      "justifyContent": "space-between",
      "paddingVertical": 10,
    }
  }
>
  <View>
    <Text>
      Soy un texto
    </Text>
  </View>
</View>
`;
